### PR TITLE
Feature/refactoring

### PR DESCRIPTION
**Pull request by unxnn**

### CODE REVIEW

File: charts/operator/README.md [hunk 1]
```diff
@@ -31,206 +31,16 @@ $ helm repo add hashicorp https://helm.releases.hashicorp.com
...
+$ helm repo update
 
 $ helm upgrade --install operator stakewise/operator \
   --namespace operator \
   --set geth.enabled=true \
   --set key=value... 
 ```
 
-**Please see the options supported in the `values.yaml` file.**
+**Please see the options supported in the `values.yaml` file.** And the full documentation at https://docs.stakewise.io
```
Comment (tsudmi) [inline charts/operator/README.md:42]:
```suggestion
**Please see the options supported in the `values.yaml` file.** The full documentation is available at https://docs.stakewise.io/node-operator/stakewise-infra-package/usage
```

File: charts/operator/templates/_helpers.tpl [hunk 2]
```diff
@@ -101,21 +101,13 @@ Write current validator info to /data directory
...
-- "--beacon-node-api-endpoint={{ $.Values.beaconChainRpcEndpoint }}"
-{{- end }}
-{{- else }}
-{{- if eq $.Values.type "prysm" }}
-- "--beacon-rpc-provider=operator-prysm:4000"
-{{- else if eq $.Values.type "lighthouse" }}
-- "--beacon-nodes=http://operator-lighthouse:5052"
-{{- else if eq $.Values.type "teku" }}
-- "--beacon-node-api-endpoint=http://operator-teku:5051"
+{{- with (first $.Values.beaconChainRpcEndpoints) }}
```
Comment (tsudmi) [inline charts/operator/templates/_helpers.tpl:109]: Teku doesn't support multiple beacon chain endpoints? Maybe check whether there is only one value provided and throw error if not?

File: charts/teku/values.yaml [hunk 3]
```diff
@@ -166,94 +166,46 @@ persistence:
...
+dataStorageArchiveFrequency: "2048"
+# Sets the strategy for handling historical chain data
+# (Valid values: ARCHIVE, PRUNE)
+dataStorageMode: PRUNE
+# Store non-canonical blocks
+dataStorageNonCanonicalBlocksEnabled: false
+# The initial state. This value should be a file or
+# URL pointing to an SSZ-encoded finalized
+# checkpoint state
+initialState: ""
```
Comment (tsudmi) [inline charts/teku/values.yaml:190]: I think this could be specified through the extraArgs

File: charts/teku/values.yaml [hunk 4]
```diff
@@ -166,94 +166,46 @@ persistence:
...
-    # allow any origin
-    corsOrigins:
-      - "*"
-    # Enable swagger-docs and swagger-ui endpoints
-    docsEnabled: false
+# URLs for Eth1 nodes
+eth1Endpoints: []
+
+# JAVA_OPTS
+javaOpts: "-Xmx4g"
```
Comment (tsudmi) [inline charts/teku/values.yaml:173]: I suggest we move the flags you shouldn't override like "dataBeaconPath" to the statefulset. Other flags that we override, but could be changed by the users we can specify in extraFlags. The flags that are equal to default ones we can omit. It's better to keep it similar to lighthouse and prysm.

File: charts/teku/templates/statefulset.yaml [hunk 5]
```diff
@@ -45,45 +45,33 @@ spec:
...
-            - --rest-api-cors-origins={{ .Values.teku.restApi.corsOrigins | join "," }}
-            - --rest-api-docs-enabled={{ .Values.teku.restApi.docsEnabled }}
+            - --data-beacon-path={{ .Values.dataBeaconPath }}
+            - --data-path={{ .Values.dataPath }}
+            - --data-storage-archive-frequency={{ .Values.dataStorageArchiveFrequency }}
+            - --data-storage-mode={{ .Values.dataStorageMode }}
+            - --data-storage-non-canonical-blocks-enabled={{ .Values.dataStorageNonCanonicalBlocksEnabled }}
+            - --eth1-endpoints={{ .Values.eth1Endpoints | join "," }}
+          {{- if .Values.restApi.enabled }}
+            - --rest-api-enabled={{ .ValuesrestApi.enabled }}
```
Comment (tsudmi) [inline charts/teku/templates/statefulset.yaml:66]:
```suggestion
            - --rest-api-enabled={{ .Values.restApi.enabled }}
```

File: charts/teku/templates/statefulset.yaml [hunk 6]
```diff
@@ -45,45 +45,33 @@ spec:
...
+            - --rest-api-host-allowlist={{ .ValuesrestApi.hostAllowList | join "," }}
```
Comment (tsudmi) [inline charts/teku/templates/statefulset.yaml:67]:
```suggestion
            - --rest-api-host-allowlist={{ .Values.restApi.hostAllowList | join "," }}
```

File: charts/teku/templates/statefulset.yaml [hunk 7]
```diff
@@ -45,45 +45,33 @@ spec:
...
+            - --rest-api-interface={{ .ValuesrestApi.interface }}
```
Comment (tsudmi) [inline charts/teku/templates/statefulset.yaml:68]:
```suggestion
            - --rest-api-interface={{ .Values.restApi.interface }}
```

File: charts/teku/templates/statefulset.yaml [hunk 8]
```diff
@@ -92,39 +80,17 @@ spec:
...
-            - --p2p-discovery-enabled={{ .Values.teku.p2p.discoveryEnabled }}
-            {{- if .Values.teku.p2p.discoveryBootNodes }}
-            - --p2p-discovery-bootnodes={{ .Values.teku.p2p.discoveryBootNodes | join "," }}
-            {{- end }}
-            {{- if .Values.teku.p2p.staticPeers }}
-            - --p2p-static-peers={{ .Values.teku.p2p.staticPeers | join "," }}
-            {{- end }}
-            - --p2p-advertised-ip={{ .Values.teku.p2p.advertisedIP }}
-            - --p2p-subscribe-all-subnets-enabled={{ .Values.teku.p2p.subscribeAllSubnets }}
+          {{- range .Values.extraFlags }}
```
Comment (tsudmi) [inline charts/teku/templates/statefulset.yaml:83]: extraFlags is missing in values.yaml

File: charts/teku/values.yaml (hunk 4 above)
Comment (tsudmi) [inline charts/teku/values.yaml:173]: I suggest we move the variables that should not be changed (dateaBeaconPath) to the statefulset, the variables that we override to extraFlags and remove everything that is equal to the default value (they can be overridden in extraFlags as well).

**Review by tsudmi — COMMENTED** [review]
- Lets rename chart from operator to validators
- Please also test whether chart deploys without issues for prysm/lighthouse/teku

File: charts/operator/templates/_helpers.tpl (hunk 2 above)
Comment (unxnn) [reply charts/operator/templates/_helpers.tpl]: Yes, it doesn't support, added additional validations.

File: charts/validators/templates/validate.yaml [hunk 9]
```diff
@@ -0,0 +1,11 @@
+{{- if and (eq .Values.network "gnosis") (eq .Values.type "teku") }}
+{{- fail "Gnosis chain network can't be used with teku client" }}
+{{- end }}
+
+{{- if not .Values.beaconChainRpcEndpoints }}
+{{- fail ".Values.beaconChainRpcEndpoints is empty" }}
+{{- end }}
+
+{{- if not .Values.vaultAddr }}
```
Comment (tsudmi) [inline charts/validators/templates/validate.yaml:9]:
```suggestion
{{- if not .Values.vaultEndpoint }}
```

File: charts/validators/README.md [hunk 10]
```diff
@@ -0,0 +1,45 @@
...
+  * Kubernetes 1.18+ - This is the earliest version of Kubernetes tested. Charts may work with earlier versions but it is untested.
+  * PV provisioner support in the underlying infrastructure
+
+## Usage
+
+To install the latest version of this chart, add dependent helm repositories
+and run `helm install`:
+
+```console
+$ helm repo add hashicorp https://helm.releases.hashicorp.com
```
Comment (tsudmi) [inline charts/validators/README.md:30]: not needed

File: charts/validators/README.md [hunk 11]
```diff
@@ -0,0 +1,45 @@
+# README
+
+> :warning: **Please note**: We take StakeWise's security and our users' trust very seriously. If 
+you believe you have found a security issue in StakeWise Helm charts, _please responsibly disclose_ 
+by contacting us at [security@stakewise.io](mailto:security@stakewise.io).
+
+This repository contains the official StakeWise Helm charts for installing
```
Comment (tsudmi) [inline charts/validators/README.md:7]: Please go through the readme and clean it up only for the validators chart

File: charts/teku/values.yaml [hunk 12]
```diff
@@ -166,94 +166,44 @@ persistence:
...
-    docsEnabled: false
+# URLs for Eth1 nodes
+eth1Endpoints: []
+
+## Extra flags for teku beacon chain node
+##
+extraFlags: []
+
+# Path to beacon node data
+dataBeaconPath: /data/teku/beacon
```
Comment (tsudmi) [inline charts/teku/values.yaml:177]: Is dataPath, dataBeaconPath supposed to be overridden? If not, I suggest we hardcode them

File: charts/validators/templates/validate.yaml (hunk 9 above)
Comment (tsudmi) [reply charts/validators/templates/validate.yaml:9]: Please search through the chart to make sure there are no other vaultAddr

**Review by tsudmi — APPROVED** [review]

**Pull request merged**